src/RTCPeerConnection.ts

```diff
@@ -552,14 +546,13 @@ export default class RTCPeerConnection extends defineCustomEventTarget(...PEER_C
 
             log.debug(`${this._pcId} onremovetrack`);
 
-            // Based on the W3C specs https://w3c.github.io/webrtc-pc/#dom-rtcpeerconnection-removetrack,
-            // we need to remove the track from any media streams
-            // that were previously passed to the `track` event.
+            // As per the spec, all we need to do is mark the track as muted:
+            // https://w3c.github.io/webrtc-pc/#process-remote-track-removal
             for (const stream of this._remoteStreams.values()) {
-                const trackIdx = stream._tracks.findIndex(t => t.id === ev.trackId);
+                const [ track ] = stream._tracks.filter(t => t.id === ev.trackId);
 
-                if (trackIdx !== -1) {
-                    stream._tracks.splice(trackIdx, 1);
+                if (track) {
+                    track._muted = true;
```

Comment:
Is the track collected anytime?

Reply:
According to the spec (linked in the commit) the track should remain there, but be marked as muted.